learn_django/trip_project/models.py

```diff
@@ -0,0 +1,27 @@
+from django.conf import settings
+from django.db import models
+
+class Trip(models.Model):
+    trip_title = models.CharField(max_length=60)
+    trip_description = models.TextField()
+    start_date = models.DateField()
+    end_date = models.DateField()
+    price = models.IntegerField()
+    created_date = models.DateField(auto_now_add=True)
+    added_by = models.ForeignKey(settings.AUTH_USER_MODEL,
+                                 null=True, blank=True, on_delete=models.SET_NULL)
+    host_name = models.CharField(max_length=100, null=True)
+    host_number = models.CharField(max_length=20, null=True)
+    poster = models.ImageField(null=True)
+    start_location = models.CharField(max_length=100, null=True)
+    destination_location = models.CharField(max_length=100, null=True)
+    destination_date = models.DateField(null=True)
+    trip_policy = models.TextField(null=True)
+    total_days = models.IntegerField(null=True)
+
+class Itenrary(models.Model):
+    trip = models.ForeignKey('Trip', on_delete=models.CASCADE)
+    Day_number = models.CharField(max_length=3)
+    date = models.DateField(null=True)
```

Comment:
how this date is different from the start_date and end_date in trip model? 

Why we need this field?

Reply:
This is for complete trip plan, the small places we will be covering during trip
e.g. we will be visiting saif-ul-malook lake on second day of trip.

Reply:
Having `date` is redundant. 

We have `trip.start_date` and `trip.end_date` and also we have `trip.total_days` these fields can be used to populate the frontend. e.g.

```
total_days = 10
start_date = 01-01-2020

lets loop from index, date to start_date+total_days:
    index: date : trip.itenrary().all().get_day(index)


```

So I would suggest removing the `date` field from here.